### DISCUSSION TITLE
Ability to configure NetworkTransforms to send as often as possible.

### DISCUSSION THREAD

**Discussion by AlphaLul** [FirstGearGames/FishNet#343] [Feature Request]
I think it would be nice to be able to configure certain NetworkTransforms to send data as often as possible, similar to how Mirror's NetworkTransform works. This is useful because things such as aiming in top-down shooters often only look good if the rotation logic is in Update. However, since NetworkTransforms send at a fixed rate that is often lower than the Update rate, it can look delayed on observing clients.

It would be nice to have a client-authoritative NT that can just send values to the server right when they update. This obviously wouldn't be the recommended way to use the NetworkTransform since it takes up more bandwidth, but it would be a useful feature to have in cases where there aren't very many good alternatives.

**Comment by maxkratt** [comment]
Wouldn't an increased tick rate help in that case?

**Reply by AlphaLul** [reply]
Not really, since as often as possible is different for each client.

**Comment by FirstGearGames** [comment]
This is not something I would include. You can disable synchronizing a
property or even axes of that property and do it manually if you like.

**Reply by AlphaLul** [reply]
Is there a way for me to send anything as fast as possible with built-in FishNet features? From what I can tell, SyncVars also send at a fixed rate, so I'm not sure I'd be able to implement it with that.

**Reply by FirstGearGames** [reply]
> Is there a way for me to send anything as fast as possible with built-in FishNet features? From what I can tell, SyncVars also send at a fixed rate, so I'm not sure I'd be able to implement it with that.

Per tick is 'fast as possible', at least if you are following the network timings. NTs already send per tick, and SyncTypes can as well by setting interval to 0f.

If you want to send certain things immediately outside the tick system, you could by changing the Timing Type on TimeManager from Tick to Variable. Note: NT and SV still will only send on tick.

**Comment by FirstGearGames** [comment]
PS: marking as declined because the current behavior is not going to change.
